secret-service/pkg/backend/scopebackend_k8s.go

```diff
@@ -0,0 +1,19 @@
+package backend
+
+import "sort"
+
+func (k K8sSecretBackend) GetScopes() ([]string, error) {
+	scopes, err := k.ScopesRepository.Read()
+	if err != nil {
+		return nil, err
+	}
+	scopeArray := make([]string, len(scopes.Scopes))
```

Comment:
With my suggestion below, you don't need to pre-create the array.
```suggestion
	var scopeArray []string
```

See https://go.dev/tour/moretypes/15 for more details

Reply:
Pre-creating the slice is needed. Behind a slice, there is an array. If the array is full, a bigger array is created with a complexity of O(N).

Reply:
> Pre-creating the slice is needed. Behind a slice, there is an array. If the array is full, a bigger array is created with a complexity of O(N).

You are correct about the negative performance implications of my approach.
However, we're talking about scopes here. There will be maybe 5 to 20 scopes, not 100k, therefore the complexity of O(N) is negligible.

Nevertheless, if you feel that you need to pre-create the slice, that's fine.